### PR TITLE
[receiver/dockerstats] Fix memory usage/percent calculation

### PR DESCRIPTION
**Pull request by gsanchezgavier**
**Description:** <Describe what has changed.>

This PR implements the following proposals:

* `container.memory.usage.total`: The proposal is to support both versions of cgroups and use the same method of the [docker cli](https://github.com/docker/cli/blob/d4f2609ce8fe63dcfca862677a50fb6c184e072f/cli/command/container/stats_helpers.go#L227) is using which represents better usage for some use-cases according to [this](https://github.com/moby/moby/issues/40727) discussion. 

* `container.memory.percent`: Similar proposal to use the same method of the [docker cli](https://github.com/docker/cli/blob/d4f2609ce8fe63dcfca862677a50fb6c184e072f/cli/command/container/stats_helpers.go#L251), based on the total usage previously calculated.

**Link to tracking Issue:** <Issue number if applicable>
#21097 

**Testing:** <Describe what testing was performed and which tests were added.>
I extended the current metrics tests. 

Output from `docker stats` of the added test:
```sh
CONTAINER ID   NAME                    CPU %     MEM USAGE / LIMIT   MEM %     NET I/O       BLOCK I/O    PIDS
f97ed5bca0a5   compassionate_mcnulty   0.00%     223.8MiB / 256MiB   87.41%    1.58kB / 0B   168kB / 0B   1
```

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: receiver/dockerstatsreceiver/receiver.go
Comment (jamesmoessis) [inline receiver/dockerstatsreceiver/receiver.go:298]: Since you've taken this function from the Docker CLI code, which is under the Apache license, it will need a copyright notice similar to `calculateCPUPercent()`.

File: receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml
Comment (jamesmoessis) [inline receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml:565]: The description of this metric is now a bit misleading, can you change that in `metadata.yml`?

It can probably just be "Total memory usage of the container".

**Review by jamesmoessis — COMMENTED** [review]
Looks good to me, just a couple of minor things to address

File: receiver/dockerstatsreceiver/receiver.go
Comment (gsanchezgavier) [reply receiver/dockerstatsreceiver/receiver.go:298]: I have updated the notice that is on top of `calculateCPUPercent` which is copied from the same place to avoid having this notice multiple times, perhaps I can move all these three functions `calculateCPUPercent`, `calculateMemUsageNoCache`, and `calculateMemoryPercent` to a single file with one notice in the top WDYT?

File: receiver/dockerstatsreceiver/receiver.go
Comment (gsanchezgavier) [reply receiver/dockerstatsreceiver/receiver.go:312]: yes `total_inactive_file` is [not present with cgroups v2](https://github.com/moby/moby/blob/7103efac9d737e8b202126e8c8e2227805e70771/daemon/stats_unix.go#L195)

File: receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml
Comment (gsanchezgavier) [reply receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml:565]: Perhaps replacing the `total cache` ?:
`Memory usage of the container. This excludes the inactive file memory.`

Or just mentioning that excludes "cache" like Docker CLI [docs](https://docs.docker.com/engine/reference/commandline/stats/#description) does in the note?
`Memory usage of the container. This excludes the cache.`

WDYT

File: receiver/dockerstatsreceiver/receiver.go
Comment (jamesmoessis) [reply receiver/dockerstatsreceiver/receiver.go:298]: I like that idea

File: receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml
Comment (jamesmoessis) [reply receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml:565]: Either one works for me

File: receiver/dockerstatsreceiver/receiver.go
Comment (gsanchezgavier) [reply receiver/dockerstatsreceiver/receiver.go]: done in [6f3b680](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21166/commits/6f3b680abdecfa295ede3e81d0c61909a13c61c7)

File: receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml
Comment (gsanchezgavier) [reply receiver/dockerstatsreceiver/testdata/mock/two_containers/expected_metrics.yaml:565]: done in [a756251](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21166/commits/a75625150da0a1b69236296dc7356a66b9ceb770)

**Review by jamesmoessis — APPROVED** [review]

**Comment by jamesmoessis** [comment]
Hi @mx-psi - this has had codeowner approval for a couple of weeks now. It needs an approver or maintainer to look at it for it to get through.

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
> Hi @mx-psi - this has had codeowner approval for a couple of weeks now. It needs an approver or maintainer to look at it for it to get through.

Woops, totally slipped through the cracks, thanks for the ping! It looks good to me so we can merge :)

**Comment by mx-psi** [comment]
This broke the lint job, presumably because #21384 was not present on this branch. #21732 should fix things.

**Pull request merged**
